src/main/java/htsjdk/samtools/cram/structure/Container.java

```diff
@@ -59,6 +63,27 @@
      */
     public long offset;
 
+
+    /**
+     * Get this container's serializable sequence ID:
+     * 0 or greater for a single reference
+     * -1 for unmapped slice
+     * -2 for multiple reference slice
```

Comment:
Can this be narrowed so it only returns valid (actual) sequence IDs ? I think having it return an int that is overloaded with the slice type sentinel values diminishes the value of having a type-safe `getSliceType` method in the first place. We want to encourage callers to use `getSliceType` *instead* of the sequence ID to determine what type of Slice it is, and to determine whether or not it has a valid sequence ID (or valid alignment span), and only call `getSequenceID` when they want/expect an actual legitimate sequenceID value. I would go so far as to have this throw if the `sliceType` isn't `SINGLE_REFERENCE`. We might have to change some other call sites that depend on the current overloading of these values.

Reply:
We do need a method to return the literal value for serialization, as required by the spec.

How about something like this: `public getValidSequenceId()` which checks for validity, and a separate `getSequenceIdForHeader()` which can return -1 and -2?  The later can eventually be `private` when we refactor `Container` to handle its own I/O.

Reply:
Added `getValidSequenceId()` and `getSequenceIdOrSentinel()` to both Slice and Container.